packaging/installer/install-required-packages.sh

```diff
@@ -0,0 +1,1772 @@
+			PACKAGES_NETDATA_DATABASE=1
+			;;
+
+		firehol|update-ipsets|firehol-all|fireqos)
+			PACKAGES_IPRANGE=1
+			PACKAGES_FIREHOL=1
+			PACKAGES_FIREQOS=1
+			PACKAGES_IPRANGE=1
+			PACKAGES_UPDATE_IPSETS=1
+			;;
+
+		demo|all)
+			PACKAGES_NETDATA=1
+			PACKAGES_NETDATA_NODEJS=1
+			PACKAGES_NETDATA_PYTHON=1
+			PACKAGES_NETDATA_PYTHON3=1
+			PACKAGES_NETDATA_PYTHON_MYSQL=1
+			PACKAGES_NETDATA_PYTHON_POSTGRES=1
+			PACKAGES_NETDATA_PYTHON_MONGO=1
+			PACKAGES_DEBUG=1
+			PACKAGES_IPRANGE=1
+			PACKAGES_FIREHOL=1
+			PACKAGES_FIREQOS=1
+			PACKAGES_UPDATE_IPSETS=1
+			PACKAGES_NETDATA_DEMO_SITE=1
+			PACKAGES_NETDATA_DATABASE=1
+			;;
+
+		help|-h|--help)
+			usage
+			exit 1
+			;;
+
+		*)
+			echo >&2 "ERROR: Cannot understand option '${1}'"
+			echo >&2 
+			usage
+			exit 1
+			;;
+	esac
+	shift
+done
+
+# Check for missing core commands like grep, warn the user to install it and bail out cleanly
+if ! command -v grep > /dev/null 2>&1; then
+	echo >&2
+	echo >&2 "ERROR: 'grep' is required for the install to run correctly and was not found on the system."
+	echo >&2 "Please install grep and run the installer again."
+	echo >&2
+	exit 1
+fi
+
+if [ -z "${package_installer}" -o -z "${tree}" ]
+	then
+	if [ -z "${distribution}" ]
+		then
+		# we dont know the distribution
+		autodetect_distribution || user_picks_distribution
+	fi
+
+	# When no package installer is detected, try again from distro info if any
+	if [ -z "${package_installer}" ]; then
+		detect_package_manager_from_distribution "${distribution}"
+	fi
+
+fi
+
+pv=$(python --version 2>&1)
+if [[ "${pv}" =~ ^Python\ 2.* ]]
+then
+	pv=2
+elif [[ "${pv}" =~ ^Python\ 3.* ]]
+then
+	pv=3
+	PACKAGES_NETDATA_PYTHON3=1
+else
+	pv=2
+fi
+
+[ "${detection}" = "/etc/os-release" ] && cat <<EOF
+
+/etc/os-release information:
+NAME            : ${NAME}
+VERSION         : ${VERSION}
+ID              : ${ID}
+ID_LIKE         : ${ID_LIKE}
+VERSION_ID      : ${VERSION_ID}
+EOF
+
+cat <<EOF
+
+We detected these:
+Distribution    : ${distribution}
+Version         : ${version}
+Codename        : ${codename}
+Package Manager : ${package_installer}
+Packages Tree   : ${tree}
+Detection Method: ${detection}
+Default Python v: ${pv} $([ ${pv} -eq 2 -a ${PACKAGES_NETDATA_PYTHON3} -eq 1 ] && echo "(will install python3 too)")
+
+EOF
+
+PACKAGES_TO_INSTALL=( $(packages | sort -u) )
+
+if [ ${#PACKAGES_TO_INSTALL[@]} -gt 0 ]
+	then
+	echo >&2
+	echo >&2 "The following command will be run:"
+	echo >&2
+	DRYRUN=1
+	${package_installer} "${PACKAGES_TO_INSTALL[@]}"
+	DRYRUN=0
+	echo >&2
+	echo >&2
+
+	if [ ${DONT_WAIT} -eq 0 -a ${NON_INTERACTIVE} -eq 0 ]
+		then
+		read -p "Press ENTER to run it > " || exit 1
+	fi
+
+	${package_installer} "${PACKAGES_TO_INSTALL[@]}" || install_failed $?
+
+	echo >&2
+	echo >&2 "All Done! - Now proceed to the next step."
+	echo >&2
+
+else
+	echo >&2
+	echo >&2 "All required packages are already installed. Now proceed to the next step."
+	echo >&2
+fi
+
+remote_log "OK"
+
+exit 0
```

Comment:
This script was directly copied from somewhere else right? Can you show a `diff` of the two here?

Reply:
Source is https://raw.githubusercontent.com/netdata/netdata-demo-site/master/install-required-packages.sh

The scripts themselves are currently identical.